### PR TITLE
fix(storage): fix condition of compaction task validator

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

When the number of files or levels in a compact task is too large, no other limitations are considered. Data flow on lsm will be smoother.

~Otherwise, with some small ssts, it may not be possible to select a sufficient sst size to guarantee to base compaction execution. (The strategy is to ensure that the number of files selected does not exceed the limit, and at the same time the size is larger than the target.)~

After the move state table, the sub level is directly inserted into the new cg. Thereafter, the base compaction picker selects a task that does not satisfy the select_level_size > base_level target level size limit based on the max_size limit (write-amp).

The reason for this is that the max_size of the picker is exactly equal to the max_base_level_size, and the size of the selected task will be smaller than the max_base_level_size, so it will not be able to satisfy the write-amp limit. This pr fixes this problem.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/meta/src/hummock/compaction/picker/compaction_task_validator.rs [hunk 1]
```diff
@@ -89,19 +89,25 @@ struct TierCompactionTaskValidationRule {
 
 impl CompactionTaskValidationRule for TierCompactionTaskValidationRule {
     fn validate(&self, input: &CompactionInput, stats: &mut LocalPickerStatistic) -> bool {
+        // Limit sstable file count to avoid using too much memory.
+        let overlapping_max_compact_file_numer = std::cmp::min(
+            self.config.level0_max_compact_file_number,
+            MAX_COMPACT_LEVEL_COUNT as u64,
+        );
+
+        if input.total_file_count >= overlapping_max_compact_file_numer
```
Comment (zwang28) [inline src/meta/src/hummock/compaction/picker/compaction_task_validator.rs:98]: This makes subsequent line `input.total_file_count <= overlapping_max_compact_file_numer` useless (always true)

Comment (Li0k) [reply src/meta/src/hummock/compaction/picker/compaction_task_validator.rs:98]: ok, we can replace `input.total_file_count <= overlapping_max_compact_file_numer` with `true`.

I'm not quite sure that if we can directly receive the task that `total_file_count` > MAX_COMPACT_LEVEL_COUNT .How did you consider this ?

Comment (zwang28) [reply src/meta/src/hummock/compaction/picker/compaction_task_validator.rs:98]: Remove this redundant code and rest LGTM.

**Review by zwang28 — APPROVED** [review]

File: src/meta/src/hummock/compaction/picker/base_level_compaction_picker.rs [hunk 2]
```diff
@@ -137,8 +137,8 @@ impl LevelCompactionPicker {
             min_compaction_bytes,
             // divide by 2 because we need to select files of base level and it need use the other
             // half quota.
-            std::cmp::min(
-                self.config.max_bytes_for_level_base,
+            std::cmp::max(
+                (self.config.max_bytes_for_level_base as f64 * 1.2) as u64,
```
Comment (Little-Wallace) [inline src/meta/src/hummock/compaction/picker/base_level_compaction_picker.rs:141]: `max_compaction_bytes/2` will be larger than `self.config.max_bytes_for_level_base`. So this change makes no difference

File: src/meta/src/hummock/compaction/picker/compaction_task_validator.rs [hunk 3]
```diff
@@ -89,26 +89,26 @@ struct TierCompactionTaskValidationRule {
...
-        );
-
-        let waiting_enough_files = {
-            if input.select_input_size > max_compaction_bytes {
-                false
-            } else {
-                input.total_file_count <= overlapping_max_compact_file_numer
-            }
-        };
+        let waiting_enough_files = input.select_input_size < max_compaction_bytes;
```
Comment (Little-Wallace) [inline src/meta/src/hummock/compaction/picker/compaction_task_validator.rs:111]: This line can be merge with line 117

**Review by Little-Wallace — APPROVED** [review]
rest LGTM

**Pull request merged**